### PR TITLE
Add created asset index to txn status

### PR DESCRIPTION
**Pull request by justicz**
This PR adds a new field to `v1.Transaction` called `TransactionResults`. This field is a struct containing `CreatedAssetIndex`, the index (if any) of an asset created by this transaction.

It computes the created asset index by looking up the block associated with the transaction if it was confirmed (and the correct type of transaction), computing the offset within that block, and adjusting based on `TxnCounter`.

Currently we're just returning this on a best-effort basis. If we don't have access to the block, we can't compute the created asset index based on just the transaction id.

### CODE REVIEW

File: daemon/algod/api/server/v1/handlers/handlers.go [hunk 1]
```diff
@@ -216,17 +217,74 @@ func assetFreezeTxEncode(tx transactions.Transaction, ad transactions.ApplyData)
...
 	s, err := txEncode(tr.Txn.Txn, tr.ApplyData)
 	if err != nil {
 		err = decorateUnknownTransactionTypeError(err, tr)
 		return v1.Transaction{}, err
 	}
 	s.ConfirmedRound = uint64(tr.ConfirmedRound)
 	s.PoolError = tr.PoolError
+	s.TransactionResults = &v1.TransactionResults{
+		CreatedAssetIndex: computeAssetIndexFromTxn(tr, l),
+	}
```
Comment (rotemh) [inline daemon/algod/api/server/v1/handlers/handlers.go]: Seems like we're enforcing behavior for future callers. i.e. all future functions must return the empty value in order to be omitted. I believe that the addition of extra information should happen at the caller level.

**Comment by rotemh** [comment]
one comment, otherwise, looks good to me

File: daemon/algod/api/server/v1/handlers/handlers.go (hunk 1 above)
Comment (justicz) [reply daemon/algod/api/server/v1/handlers/handlers.go]: fixed :+1:

File: daemon/algod/api/server/v1/handlers/handlers.go [hunk 2]
```diff
@@ -682,12 +738,17 @@ func PendingTransactionInformation(ctx lib.ReqContext, w http.ResponseWriter, r
...
 	if txn, ok := ctx.Node.GetPendingTransaction(txID); ok {
+		ledger := ctx.Node.Ledger()
 		responseTxs, err := txWithStatusEncode(txn)
 		if err != nil {
 			lib.ErrorResponse(w, http.StatusInternalServerError, err, errFailedToParseTransaction, ctx.Log)
 			return
 		}
 
+		responseTxs.TransactionResults = &v1.TransactionResults{
+			CreatedAssetIndex: computeAssetIndexFromTxn(txn, ledger),
```
Comment (rotemh) [inline daemon/algod/api/server/v1/handlers/handlers.go]: can you just add the comment that it will be omitted if there is no assetID?

Comment (justicz) [reply daemon/algod/api/server/v1/handlers/handlers.go]: done :+1:

**Comment by algobolson** [comment]
This is probably good but I feel like it would be better with a test that submits a create-asset transaction, gets this response, then looks up the creator's account info to see that the same asset id is listed there.

**Pull request merged**